src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectTemplate.cs

```diff
@@ -3,217 +3,159 @@
-
-        private IReadOnlyDictionary<string, ICacheTag> _tags;
-
-        private IReadOnlyDictionary<string, ICacheParameter> _cacheParameters;
-
-        private IReadOnlyList<ITemplateParameter> _parameters;
-
-        internal RunnableProjectTemplate(JObject raw, IGenerator generator, IFile configFile, IRunnableProjectConfig config, IFile localeConfigFile, IFile hostConfigFile)
+        private readonly IRunnableProjectConfig _config;
+        private readonly IFile _configFile;
+        private readonly IGenerator _generator;
+        private readonly IFile _localeConfigFile;
+        private readonly IFile _hostConfigFile;
+
+        internal RunnableProjectTemplate(
+            JObject raw,
+            IGenerator generator,
+            IFile configFile,
+            IRunnableProjectConfig config,
+            IFile localeConfigFile,
+            IFile hostConfigFile)
         {
-            config.SourceFile = configFile;
-            ConfigFile = configFile;
-            Generator = generator;
-            Source = configFile.MountPoint;
-            Config = config;
-            DefaultName = config.DefaultName;
-            Name = config.Name;
-            Identity = config.Identity ?? config.Name;
-
-            ShortNameList = config.ShortNameList ?? new List<string>();
-
-            Author = config.Author;
-            Tags = config.Tags ?? new Dictionary<string, ICacheTag>(StringComparer.OrdinalIgnoreCase);
-            CacheParameters = config.CacheParameters ?? new Dictionary<string, ICacheParameter>(StringComparer.OrdinalIgnoreCase);
-            Description = config.Description;
-            Classifications = config.Classifications;
-            GroupIdentity = config.GroupIdentity;
-            Precedence = config.Precedence;
-            LocaleConfigFile = localeConfigFile;
-            IsNameAgreementWithFolderPreferred = raw.ToBool("preferNameDirectory", false);
-            HostConfigPlace = hostConfigFile?.FullPath;
-            ThirdPartyNotices = raw.ToString("thirdPartyNotices");
+            _config = config;
+            _configFile = configFile;
+            _generator = generator;
+            _localeConfigFile = localeConfigFile;
+            _hostConfigFile = hostConfigFile;
             _raw = raw;
-            BaselineInfo = config.BaselineInfo;
+            config.SourceFile = configFile;
         }
 
-        public IDirectory TemplateSourceRoot
+        IDirectory ITemplate.TemplateSourceRoot
         {
             get
             {
-                return ConfigFile?.Parent?.Parent;
+                return _configFile?.Parent?.Parent;
             }
         }
 
-        public string Identity { get; }
+        string ITemplateInfo.Identity => _config.Identity ?? _config.Name;
 
-        public Guid GeneratorId => Generator.Id;
+        Guid ITemplateInfo.GeneratorId => _generator.Id;
 
-        public string Author { get; }
+        string ITemplateInfo.Author => _config.Author;
```

Comment:
Do we need to implement these explicitly? Can we do
```
public string Author => _config.Author:
```

Reply:
Yes, I made it on purpose. `RunnableProjectTemplate` mainly defines what `ITemplateInfo` and `ITemplate` will be for this format. I don't want it to be used differently. More refactoring will follow here as I would like to remove casting ITemplate/ITemplateInfo to RunnableProjectTemplate which is frequent in this generator.